tests/socket_test.c

```diff
@@ -432,6 +436,30 @@ static int s_test_tcp_socket_communication(struct aws_allocator *allocator, void
 
 AWS_TEST_CASE(tcp_socket_communication, s_test_tcp_socket_communication)
 
+#if defined(USE_VSOCK)
+static int s_test_vsock_loopback_socket_communication(struct aws_allocator *allocator, void *ctx) {
+/* Without vsock loopback it's difficult to test vsock functionality.
+ * Also note that having this defined does not guarantee that it's available
+ * for use and there's no path to figure out dynamically if it can be used. */
+#    if defined(VMADDR_CID_LOCAL)
+    (void)ctx;
+
+    struct aws_socket_options options;
+    AWS_ZERO_STRUCT(options);
+    options.connect_timeout_ms = 3000;
+    options.type = AWS_SOCKET_STREAM;
+    options.domain = AWS_SOCKET_VSOCK;
+
+    struct aws_socket_endpoint endpoint = {.address = "1" /* VMADDR_CID_LOCAL */, .port = 8127};
+
+    return s_test_socket(allocator, &options, &endpoint);
+#    endif
+    return 0;
```

Comment:
trivial: maybe like this, to avoid the compiler seeing 2 return statements one after the other? Pretty sure some linter somewhere would be mad about that:
```
#else
    return 0;
#endif
```